rest-api/dao/participant_summary_dao.py

```diff
@@ -336,6 +394,67 @@ def calculate_enrollment_status(self, consent_for_study_enrollment_and_ehr,
       return EnrollmentStatus.MEMBER
     return EnrollmentStatus.INTERESTED
 
+  def calculate_member_time(self, consent_for_study_enrollment_and_ehr, participant_summary):
+    if consent_for_study_enrollment_and_ehr and \
+      participant_summary.enrollmentStatusMemberTime is not None:
+      return participant_summary.enrollmentStatusMemberTime
+    elif consent_for_study_enrollment_and_ehr:
+      return participant_summary.consentForElectronicHealthRecordsTime
+    else:
+      return None
+
+  def calculate_core_stored_sample_time(self, consent_for_study_enrollment_and_ehr,
+                                        participant_summary):
+    if consent_for_study_enrollment_and_ehr and \
+      participant_summary.numCompletedBaselinePPIModules == \
+      self._get_num_baseline_ppi_modules() and \
+      participant_summary.physicalMeasurementsStatus == PhysicalMeasurementsStatus.COMPLETED and \
+      participant_summary.samplesToIsolateDNA == SampleStatus.RECEIVED:
+
+      keys = ['sampleStatus%sTime' % test
+              for test in config.getSettingList(config.DNA_SAMPLE_TEST_CODES)]
+      sample_stored_time_list = \
+        [v for k, v in participant_summary if k in keys and v is not None]
+
+      sample_stored_time = min(sample_stored_time_list) if sample_stored_time_list else None
+
+      if sample_stored_time is not None:
+        return max([sample_stored_time,
```

Comment:
I would add:

if participant_summary.enrollmentStatusCoreStoredSampleTime:
  return participant_summary.enrollmentStatusCoreStoredSampleTime 

Basically, we only want to set this field if it isn't set already. This is because the timestamps we are deriving this from get updated when repeated data comes in (e.g. we get physical measurements or the same questionnaire response twice.) See https://precisionmedicineinitiative.atlassian.net/browse/DA-607.

We should *clear* the field if all the criteria aren't met anymore -- that can happen if something gets canceled, etc. -- but we shouldn't set the field to a new value if it's already set.